cmd/boostx/stats_cmd.go

```diff
@@ -78,68 +94,124 @@ var statsCmd = &cli.Command{
+		agentVersions := make(map[string]int)
+		transportProtos := make(map[string]int)
 
-		for _, maddr := range withMinPower {
+		throttle = make(chan struct{}, cctx.Int("sp-query-concurrency"))
+		for i, maddr := range withMinPower {
 			select {
 			case <-ctx.Done():
 				return nil
 			default:
 			}
-			err := func() error {
-				addrInfo, err := cmd.GetAddrInfo(ctx, api, maddr)
-				if err != nil {
-					return fmt.Errorf("getting provider multi-address: %w", err)
-				}
 
-				log.Debugw("connecting to storage provider",
-					"id", addrInfo.ID, "multiaddrs", addrInfo.Addrs, "addr", maddr)
-
-				if err := n.Host.Connect(ctx, *addrInfo); err != nil {
-					return fmt.Errorf("connecting to peer %s: %w", addrInfo.ID, err)
-				}
+			spQueryMax := cctx.Int("sp-query-max")
+			if spQueryMax > 0 && i >= spQueryMax {
+				break
+			}
 
-				protos, err := n.Host.Peerstore().GetProtocols(addrInfo.ID)
+			throttle <- struct{}{}
+			wg.Add(1)
+			go func(maddr address.Address) {
+				defer wg.Done()
+				defer func() {
+					<-throttle
+				}()
+				err := func() error {
+					addrInfo, err := cmd.GetAddrInfo(ctx, api, maddr)
+					if err != nil {
+						return fmt.Errorf("getting provider multi-address: %w", err)
+					}
+
+					log.Debugw("connecting to storage provider",
+						"id", addrInfo.ID, "multiaddrs", addrInfo.Addrs, "addr", maddr)
+
+					if err := n.Host.Connect(ctx, *addrInfo); err != nil {
+						return fmt.Errorf("connecting to peer %s: %w", addrInfo.ID, err)
+					}
+
+					// Get peer's libp2p protocols
+					protos, err := n.Host.Peerstore().GetProtocols(addrInfo.ID)
+					if err != nil {
+						return fmt.Errorf("getting protocols for peer %s: %w", addrInfo.ID, err)
+					}
+					sort.Strings(protos)
+
+					// Get peer's libp2p agent version
+					agentVersionI, err := n.Host.Peerstore().Get(addrInfo.ID, "AgentVersion")
+					if err != nil {
+						return fmt.Errorf("getting agent version for peer %s: %w", addrInfo.ID, err)
+					}
+					agentVersion, ok := agentVersionI.(string)
+					if !ok {
+						return fmt.Errorf("AgentVersion for peer %s is not a string: type %T", addrInfo.ID, agentVersionI)
+					}
+
+					// Get SP's supported transports
+					var transports *transports_types.QueryResponse
+					if contains(protos, string(lp2pimpl.TransportsProtocolID)) {
+						client := lp2pimpl.NewTransportsClient(n.Host)
+						transports, err = client.SendQuery(ctx, addrInfo.ID)
+						if err != nil {
+							fmt.Printf("Failed to fetch transports from peer %s: %s\n", addrInfo.ID, err)
+						}
+					}
+
+					var out string
+					out += "Provider " + maddr.String()
+					if contains(protos, "/fil/storage/mk/1.2.0") {
+						out += " is running boost"
+
+						lk.Lock()
+						boostNodes++
+						boostQualityAdjPower = big.Add(boostQualityAdjPower, minerToMinerPower[maddr].QualityAdjPower)
+						boostRawBytePower = big.Add(boostRawBytePower, minerToMinerPower[maddr].RawBytePower)
+						agentVersions[agentVersion]++
```

Comment:
I think it might be beneficial to count all agent versions, not just boost versions. Right now we only have an indication of Boost versus non boost nodes, but doing a full agent version would give us a better breakdown of the non boost versions. We could just do market nodes for now, as checking the non market nodes isn't too helpful for boost planning purposes.